packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowFields.tsx

```diff
@@ -43,25 +44,35 @@ export class DetailsRowFields extends React.PureComponent<IDetailsRowFieldsProps
                 cellStyleProps.cellRightPadding +
                 (column.isPadded ? cellStyleProps.cellExtraRightPadding : 0);
 
-          const { onRender = onRenderItemColumn } = column;
+          const { onRender = onRenderItemColumn, getValueKey = getCellValueKey } = column;
           const cellContentsRender =
             cellsByColumn && column.key in cellsByColumn
               ? cellsByColumn[column.key]
               : onRender
               ? onRender(item, itemIndex, column)
               : getCellText(item, column);
 
+          // generate a key that auto-dirties when content changes, to force the container to re-render, to trigger animation
+          let key: number | string;
+          try {
+            key = getValueKey ? getValueKey(item, itemIndex, column) : getCellText(item, column) + column.key + itemIndex;
```

Comment:
That is potentially a very long string to be used as a key. `itemIndex` is unnecessary since the parent `<div>` contains only children of the same item. Also, `getValueKey` seems too specific to this use-case. There is no reason why this component needs to know about the semantics of the custom key being used.

Please do something along the lines of:
`let itemKey = this.props.getKey ? this.props.getKey(item, itemIndex, column, columnIndex) : columnIndex`

(similar how the key for a page is chosen in `List`).

Reply:
the reason called getValueKey is emphasis key is made from value therefore value change key will change.  I can remove the fallback to just column.key + itemIndex to be unique and short.

Reply:
`DetailsRowFields` should be agnostic of what type of key is going to be passed. What if some other component wants to use `DetailsRowFields`, but not pass a value as a key, but something else? I don't see any reason why we would name this new prop after this specific use case of a value changing.

Also, why is the previously used `columnIndex` not sufficient in case no `getCellKey` is passed?